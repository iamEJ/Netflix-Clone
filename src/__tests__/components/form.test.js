import React from "react";
import { render } from "@testing-library/react";
import Form from "../../components/form";

jest.mock("react-router-dom");

describe("<Form />", () => {
  it("renders <Form /> with populated data", () => {
    const { container, getByText, getByPlaceholderText } = render(
      <Form>
        <Form.Title>Sign In</Form.Title>
        <Form.Error></Form.Error>
        <Form.Base>
          <Form.Input
            placeholder="Enter email address"
            type="email"
            onChange={() => {}}
          />
          <Form.Input
            placeholder="Enter password"
            type="password"
            onChange={() => {}}
          />
          <Form.Submit type="submit" disabled>
            Sign In
          </Form.Submit>
        </Form.Base>
        <Form.Text>
          New to Netflix? <Form.Link to="/signup">Sign up now.</Form.Link>
        </Form.Text>
        <Form.TextSmall>
          This page is protected by Google reCAPTCHA to ensure you're not a bot.
          Learn more.
        </Form.TextSmall>
      </Form>
    );
    expect(
      getByText(
        "This page is protected by Google reCAPTCHA to ensure you're not a bot. Learn more."
      )
    ).toBeTruthy();
    expect(getByText("Sign Up Now.")).toBeTruthy();
    expect(getByText("Sign In")).toBeTruthy();
    expect(getByText("Sign In").disabled).toBeTruthy();
    expect(getByPlaceholderText("Enter email address")).toBeTruthy();
    expect(getByPlaceholderText("Enter password")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
