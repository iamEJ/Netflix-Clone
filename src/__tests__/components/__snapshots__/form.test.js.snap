// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> renders <Form /> with populated data 1`] = `
<div
  class="form__Container-rl5qdm-0 etyNXn"
>
  <h1
    class="form__Title-rl5qdm-3 irCIPh"
  >
    Sign In Now
  </h1>
  <div
    class="form__Error-rl5qdm-1 ikLlWu"
  />
  <form
    class="form__Base-rl5qdm-2 khUISa"
  >
    <input
      class="form__Input-rl5qdm-7 emWTFb"
      placeholder="Enter email address"
      type="email"
    />
    <input
      class="form__Input-rl5qdm-7 emWTFb"
      placeholder="Enter password"
      type="password"
    />
    <button
      class="form__Submit-rl5qdm-8 iYbwCy"
      disabled=""
      type="submit"
    >
      Sign In
    </button>
  </form>
  <p
    class="form__Text-rl5qdm-4 gWRWZU"
  >
    New to Netflix? 
    <link
      class="form__Link-rl5qdm-6 bQqFRm"
      to="/signup"
    >
      Sign up now.
    </link>
  </p>
  <p
    class="form__TextSmall-rl5qdm-5 cIzopo"
  >
    This page is protected by Google reCAPTCHA to ensure you're not a bot. Learn more.
  </p>
</div>
`;
